sapmon/payload/provider/prometheus.py

```diff
@@ -0,0 +1,169 @@
+    colTimeGenerated = None
+    excludeRegex = re.compile(r"^(?:go|promhttp|process)_")
+    lastResult = ([], None)
+
+    def __init__(self,
+                 provider: ProviderInstance,
+                 **kwargs):
+        return super().__init__(provider, **kwargs)
+
+    def _actionFetchMetrics(self,
+                            includePrefixes: str) -> bool:
+        self.tracer.info("[%s] Fetching metrics" % self.fullName)
+        metricsData = self.providerInstance.fetch_metrics()
+        self.lastResult=(metricsData, includePrefixes)
+        if metricsData == None:
+            self.tracer.info("[%s] Unable to fetch metrics" % self.fullName)
+            return False
+        return self.updateState()
+
+    # Convert last result into a JSON string (as required by Log Analytics Data Collector API)
+    def generateJsonString(self) -> str:
+        # The correlation_id can be used to group fields from the same metrics call
+        correlation_id = str(uuid.uuid4())
+        fallback_datetime=datetime.now(timezone.utc)
+
+        def prometheusSample2Dict(sample):
+            """
+            Convert a prometheus metric sample to Python dictionary for serialization
+            """
+            TimeGenerated = fallback_datetime
+            if sample.timestamp:
+                TimeGenerated = datetime.fromtimestamp(sample.timestamp, tz=timezone.utc)
+            sample.labels["instance"] = self.providerInstance.instance
+            sample_dict = {
+                "name" : sample.name,
+                "labels" : json.dumps(sample.labels, separators=(',',':'), sort_keys=True),
+                "value" : sample.value,
+                "TimeGenerated": TimeGenerated,
+                "instance": self.providerInstance.instance,
+                "correlation_id": correlation_id
+            }
+            # FIXME: Implement custom fields
+            #for (k, v) in promUrl.customFields:
+            #    sample_dict["custom_%s" % k] = v
+            return sample_dict
+
+        def filter_prometheus_metric(metric):
+            """
+            Filter out names based on our exclude and include lists
+            """
+            if includePrefixesRegex:
+                return bool(includePrefixesRegex.match(metric.name))
+            return not bool(self.excludeRegex.match(metric.name))
+
+        prometheusMetricsText = self.lastResult[0]
+        includePrefixes = self.lastResult[1]
+        # If a prefix was given it has to compile to a valid regular expression
+        if includePrefixes:
+            try:
+                includePrefixesRegex = re.compile(includePrefixes)
+            except re.error:
+                self.tracer.error("[%s] includePrefixes must be a valid regular expression: %s" %
+                                  (self.fullName, includePrefixes))
+                return False
+        resultSet = list()
+        self.tracer.info("converting result set into JSON")
```

Comment:
Adding the check name for better readability
```suggestion
        self.tracer.info("[%s] converting result set into JSON" % self.fullName)
```